src/metabase/api/user.clj

```diff
@@ -84,7 +84,7 @@
   {password     [Required ComplexPassword]}
   (check-self-or-superuser id)
   (let-404 [user (sel :one [User :password_salt :password] :id id :is_active true)]
-    (when-not (:is_superuser @*current-user*)
+    (when (= id (:id @*current-user*))
```

Comment:
previously we only checked the original password if the caller was a non-superuser (aka a non-admin updating their own account).

the change here is meant to enforce that if this is a user updating their own account (regardless of whether they are an superuser) then they must supply the existing password.

before this conditional we are already enforcing a check that ensures that either the user is a superuser (updating any account) or they are a normal user and only trying to update their own account.  so effectively we are covering everything and ensuring that normal users can only update their own account and when that's the case we check the current password, and if this is a superuser we let them update any account, but if its their own we expect that they supplied their own current password.